clippy_lints/src/methods/mod.rs

```diff
@@ -4523,16 +4546,27 @@ impl Methods {
                     }
                 },
                 ("find", [arg]) => {
-                    if let Some(("cloned", recv2, [], _span2, _)) = method_call(recv) {
-                        // if `arg` has side-effect, the semantic will change
-                        iter_overeager_cloned::check(
-                            cx,
-                            expr,
-                            recv,
-                            recv2,
-                            iter_overeager_cloned::Op::FixClosure(name, arg),
-                            false,
-                        );
+                    match method_call(recv) {
+                        Some(("cloned", recv2, [], _span2, _)) => {
+                            // if `arg` has side-effect, the semantic will change
+                            iter_overeager_cloned::check(
+                                cx,
+                                expr,
+                                recv,
+                                recv2,
+                                iter_overeager_cloned::Op::FixClosure(name, arg),
+                                false,
+                            );
+                        },
+                        Some(("enumerate", _, [], _, span2)) => {
+                            manual_position::check(cx, expr, arg, span2, false);
+                        },
+                        Some(("rev", recv2, [], _, _)) => {
+                            if let Some(("enumerate", _, [], _, span2)) = method_call(recv2) {
```

Comment:
it's very unlikely, but what happens if someone wrote `iter().rev().rev().enumerate().find()` 
or `iter().rev().enumerate().rev().find()`? 😄